resources/certificate.rb

```diff
@@ -1,26 +1,26 @@
 unified_mode true
 
 provides :certificate
+default_action :install
 
 property :certfile, String
-property :cert_password, String
-property :keychain, String
+property :cert_password, String, sensitive: true
+property :keychain, String, required: true
+property :kc_passwd, String, required: true, sensitive: true
 property :apps, Array
 
-action_class do
-  def keychain
-    new_resource.property_is_set?(:keychain) ? new_resource.keychain : ''
-  end
-end
-
 action :install do
-  cert = SecurityCommand.new(new_resource.certfile, keychain)
+  cert = SecurityCommand.new(new_resource.certfile, new_resource.keychain)
 
   execute 'unlock keychain' do
-    command Array(cert.unlock_keychain(node['macos']['admin_password']))
+    command Array(cert.unlock_keychain(new_resource.kc_passwd))
   end
 
+  cert_shasum = shell_out("shasum #{new_resource.certfile}").stdout.upcase.gsub(/\s.+/, '')
+  find_cert_output = shell_out("/usr/bin/security find-certificate -a -Z #{new_resource.keychain}").stdout
```

Comment:
`/usr/bin/security find-certificate -a -Z | awk '/SHA/ {print $3}'` might optimize the `include?` a bit?

Reply:
I believe that finds the shasums of all certificates on all keychains. So depending on how many certs/keychains there are, this more direct approach is probably faster and easier to read. Again, I don't really have any metrics to confirm or deny.